worker/worker.py

```diff
@@ -518,22 +518,34 @@ def gcc_version():
         bufsize=1,
         close_fds=not IS_WINDOWS,
     ) as p:
+        clang_major = "None"
```

Comment:
If I'm not wrong this could be simplified:
```python
        for line in iter(p.stdout.readline, ""):
            if "__GNUC__" in line or "__clang_major__" in line:
                major = line.split()[2]
            if "__GNUC_MINOR__" in line or "__clang_minor__" in line:
                minor = line.split()[2]
            if "__GNUC_PATCHLEVEL__" in line or "__clang_patchlevel__" in line:
                patchlevel = line.split()[2]

    if p.returncode != 0:
        print("g++ version query failed with return code {}".format(p.returncode))
        return None

    try:
        major = int(major)
        minor = int(minor)
        patchlevel = int(patchlevel)
```

Reply:
clang also defines __GNUC__, __GNUC_MINOR__ and __GNUC_PATCHLEVEL__, but with different values than in __clang_major__, ... . In the above suggestion the resulting values would depend on the order of the output, which is not a solid solution although it might work by random.

Reply:
On my system the clang output of "g++ -E -dM -" is:

#define __GNUC_MINOR__ 2
#define __GNUC_PATCHLEVEL__ 1
#define __GNUC_STDC_INLINE__ 1
#define __GNUC__ 4
...
#define __clang_major__ 12
#define __clang_minor__ 0
#define __clang_patchlevel__ 5
#define __clang_version__ "12.0.5 (clang-1205.0.22.9)"

Reply:
Oh, you are right, also clang/msys2 had the GNUC define
```
$ g++ -E -dM - < /dev/null | grep GNUC
#define __GNUC_MINOR__ 2
#define __GNUC_PATCHLEVEL__ 1
#define __GNUC_STDC_INLINE__ 1
#define __GNUC__ 4
```